stan/math/fwd/scal/fun/is_inf.hpp

```diff
@@ -20,6 +20,12 @@ inline int is_inf(const fvar<T>& x) {
   return is_inf(x.val());
 }
 
+// forwarding for ADL
+template <class T>
+inline auto isinf(const fvar<T>& a) {
```

Comment:
A bunch of questions:

1) why is this function here? (this is `isinf` and not `is_inf`)
2) why is it declared and defined multiple times? (in other files too)
3) why is `isinf` in the `stan::math` namespace? I'm guessing you need it in the `std` namespace?

Reply:
isinf is the name of a function that is called by Eigen and / or the complex library without a namespace prefix and with fvar as an argument. Without this definition, there would be an ADL failure / compile error. It would not have been necessary if stan had used isinf for its function name.

Reply:
I see how we've handled it elsewhere. Is there a reason we can't do the same here? By that, I mean specialize `isnan` and `isinf` for `stan::math::fvar<T>` and include it in its own header file called `stan/math/fwd/core/std_isnan.hpp` and `std_isinf`?
See:
stan/math/fwd/core/std_numeric_limits.hpp
stan/math/rev/core/std_isnan.hpp
stan/math/rev/core/std_numeric_limits.hpp
stan/math/rev/core/std_isinf.hpp

Reply:
That would not actually fix the problem. The other packages aren't always calling std::isinf. They are calling isinf with no namespace prefix. Thus, isinf must be defined in the stan::math namespace. That's just how ADL forces us to work with stan::math::fvar arguments.

Reply:
The two standard libraries and Eigen

Reply:
FYI: https://github.com/eigenteam/eigen-git-mirror/blob/777b928d701702e2c8f8fa50643688678bdd727b/Eigen/src/Core/MathFunctions.h#L794

Reply:
I think we're way past that; `std::complex` for our types are unspecified. We should do what we can verify is correct, is maintainable, and makes sense.